src/chains/tests/test_signals_ff_hook_events.py

```diff
@@ -0,0 +1,376 @@
+import responses
+from django.test import TestCase, override_settings
+from faker import Faker
+
+from ..models import Feature, Wallet
+from ..tests.factories import (
+    ChainFactory,
+    FeatureFactory,
+    GasPriceFactory,
+    WalletFactory,
+)
+
+fake = Faker()
+Faker.seed(0)
+
+
+@override_settings(FF_HOOK_EVENTS=True)
+class ChainNetworkHookTestCaseSetupCheck(TestCase):
+    @responses.activate
+    @override_settings(CGW_URL=None, CGW_FLUSH_TOKEN="example-token")
+    def test_no_cgw_call_with_no_url(self) -> None:
+        ChainFactory.create()
+
+        assert len(responses.calls) == 0
+
+    @responses.activate
+    @override_settings(CGW_URL="http://127.0.0.1", CGW_FLUSH_TOKEN=None)
+    def test_no_cgw_call_with_no_token(self) -> None:
+        ChainFactory.create()
+
+        assert len(responses.calls) == 0
+
+
+@override_settings(
+    CGW_URL="http://127.0.0.1", CGW_FLUSH_TOKEN="example-token", FF_HOOK_EVENTS=True
+)
+class ChainNetworkHookWithFFHookEventsTestCase(TestCase):
+    @responses.activate
+    def test_on_chain_create(self) -> None:
+        chain_id = fake.pyint()
+        responses.add(
+            responses.POST,
+            "http://127.0.0.1/v1/hooks/events",
+            status=200,
+            match=[
+                responses.matchers.header_matcher(
+                    {"Authorization": "Basic example-token"}
+                ),
+                responses.matchers.json_params_matcher(
+                    {"type": "CHAIN_UPDATE", "chain_id": str(chain_id)}
+                ),
+            ],
+        )
+
+        ChainFactory.create(id=chain_id)
+
+        assert len(responses.calls) == 1
+        assert isinstance(responses.calls[0], responses.Call)
+        assert responses.calls[
+            0
+        ].request.body == f'{{"type": "CHAIN_UPDATE", "chainId": "{chain_id}"}}'.encode(
+            "utf-8"
+        )
+        assert responses.calls[0].request.url == "http://127.0.0.1/v1/hooks/events"
+        assert (
+            responses.calls[0].request.headers.get("Authorization")
+            == "Basic example-token"
+        )
+
+    @responses.activate
+    def test_on_chain_delete(self) -> None:
+        chain = ChainFactory.create()
+
+        chain.delete()
+
+        # 2 calls: one for creation and one for deletion
+        assert len(responses.calls) == 2
```

Comment:
Should we test here the `CHAIN_UPDATE` webhook payload for a deletion event as we do for creation/update?

Reply:
Addressed in [bfcbbbf](https://github.com/safe-global/safe-config-service/pull/964/commits/bfcbbbfc65b52d790114b3960c1e3dce3b2e30be)